src/Passes/Source/AllocationManager/IAllocationManager.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "AllocationManager/IAllocationManager.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    IAllocationManager::~IAllocationManager() = default;
```

Comment:
Nit:
I remember the case when in my code I _had_ to specify `= default` for the destructor in the .cpp file rather than in the .hpp file. But for that case there was a strong reason that I will hardly recall right now (but I explained that in the comments).

For my curiosity can you tell why you placed `= default` for the destructor _in the .cpp file_ for this particular class? (feel free to ignore)

Reply:
This is done to prevent the virtual destructor to be defined in multiple translation units and hence causing linking errors.

Reply:
This is the first time I hear anything like that. Your words make me think that every occurrence of `= default` (applicable not only to destructors but also to the constructors and assignment operators) should be placed in the .cpp files, and never to the .hpp files. I strongly doubt that it is a rule to follow. I don't expect problems with placing `=default` to .hpp (at least for this particular case).

(Up to you, I don't mean that the code needs to be changed. I just wanted to learn in this discussion something new, or to share what I know :-) )